osutil/user.go

```diff
@@ -204,20 +211,35 @@ func AddUser(name string, opts *AddUserOptions) error {
 		return fmt.Errorf("cannot add user %q: name contains invalid characters", name)
 	}
 
-	cmdStr := []string{
-		"adduser",
-		"--force-badname",
-		"--gecos", opts.Gecos,
-		"--disabled-password",
+	var userTool string
+	var cmdStr []string
+	if hasAddUserExecutable() {
+		userTool = "adduser"
+		cmdStr = []string{
+			"adduser",
+			"--force-badname",
+			"--gecos", opts.Gecos,
+			"--disabled-password",
+		}
+	} else {
+		userTool = "useradd"
+		cmdStr = []string{
+			"useradd",
+			"--badnames",
```

Comment:
According to the man page, this is `--badname`, without the 's'... but the command help shows `--badnames`. Do both work? What should be the authoritative?

Reply:
It was corrected to singular form in https://github.com/shadow-maint/shadow/commit/45d674621918664c8736f94f862e86bddf4c3fd4 however, I still see one instance of badnames. I guess we need to survey the versions we want to support. I also usee usermod still using plural.

So it is plural on Jammy, but singular after Jammy. I wonder how much stuff will break because of this.

Maybe use short options `-b` because it is the same in all versions it seems.

Reply:
Submitted https://github.com/shadow-maint/shadow/pull/820/files

Reply:
Ouch, why on earth they didn't just keep compatibility with both singular/plural, at least for a while...

Reply:
yep, it blew my mind when I figured this changed from after Jammy. Why?